### PR TITLE
Update actions

### PR DESCRIPTION
**Pull request by simonhamp**
## What does this PR do?

- Updates action versions to their latest ones
- Fixes the dependency cache key naming logic, bringing the other environments into line with what's implemented in #420 - caches weren't being used at all when specifying an explicit set of extensions because the key contained commas

This reduces errors and improves build times.

## Checklist before merging

> If your PR involves the changes mentioned below and completed the action, please tick the corresponding option.
> If a modification is not involved, please skip it directly.

- [ ] If it's a extension or dependency update, make sure adding related extensions in `src/global/test-extensions.php`.
- [ ] If you changed the behavior of static-php-cli, add docs in [static-php/static-php-cli-docs](https://github.com/static-php/static-php-cli-docs) .
- [ ] If you updated `config/xxxx.json` content, run `bin/spc dev:sort-config xxx`.

### CODE REVIEW

**Comment by crazywhalecc** [comment]
Previously I modified [test workflow](https://github.com/crazywhalecc/static-php-cli/blob/d98d6aeffb347dceb8ed50a0ecec3ae4b530275a/.github/workflows/tests.yml#L148C1-L153C50) for dynamic caching: Just cache downlods dir again and again. When new extension added, it will automatically skip downloading other sources that has been cached.

**Comment by simonhamp** [comment]
@crazywhalecc I'm not sure I follow you. The cache in these didn't appear to be working at all because the key name was incorrect. This had the result of always downloading the extension files.

This PR fixes that so each action will now correctly cache the set of extensions that were requested.

This should make builds _of the same PHP-version+OS+extension set_ a little faster, at least during the cache's lifetime.

**Comment by simonhamp** [comment]
In case it's not clear, this dependency cache is the extension build files (the result of `bin/spc download`).

I think the variance you were seeing in `cache-hit != true` was exactly down to this problem of commas in the key name.

This is why I've hashed the details of the build into a key name that will _never_ have commas regardless of how the build is configured.

File: .github/workflows/build-macos-x86_64.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
     name: build ${{ inputs.version }} on macOS x86_64
     runs-on: macos-latest
```
Comment (crazywhalecc) [inline .github/workflows/build-macos-x86_64.yml:40]: macos-13 (macos-latest is arm64 now)

**Comment by crazywhalecc** [comment]
What I mean is that today's best caching strategies don't perform well in Actions. It has nothing to do with this PR. After fixing this small problem, we merge it into the main branch and test it again.

File: .github/workflows/build-macos-x86_64.yml (hunk 1 above)
Comment (simonhamp) [reply .github/workflows/build-macos-x86_64.yml:40]: This was already fixed in #427, which has been merged, so this should be ok.

**Review by crazywhalecc — APPROVED** [review]

**Pull request merged**